BackendIntegrationTests/BackendIntegrationTests.swift

```diff
@@ -226,7 +226,45 @@ class BackendIntegrationSK1Tests: XCTestCase {
         eligibility = await Purchases.shared.checkTrialOrIntroDiscountEligibility([productID])
         expect(eligibility[productID]?.status) == .ineligible
     }
-    
+
+    func testUserHasNoEligibleOffersByDefault() async throws {
+        let (_, created) = try await Purchases.shared.logIn(UUID().uuidString)
+        expect(created) == true
+
+        let offerings = try await Purchases.shared.offerings()
+        let product = try XCTUnwrap(offerings.current?.monthly?.storeProduct)
+
+        expect(product.discounts).to(haveCount(1))
+        expect(product.discounts.first?.offerIdentifier) == "com.revenuecat.monthly_4.99.1_free_week"
+
+        let offers = await product.getEligiblePromotionalOffers()
+        expect(offers).to(beEmpty())
+    }
+
+    func testPurchaseWithDiscount() async throws {
+        let (_, created) = try await Purchases.shared.logIn(UUID().uuidString)
+        expect(created) == true
+
+        let offerings = try await Purchases.shared.offerings()
+        let product = try XCTUnwrap(offerings.current?.monthly?.storeProduct)
+
+        try await self.purchaseMonthlyOffering()
+        let entitlement1 = try self.verifyEntitlementWentThrough()
+
+        print(entitlement1)
+
+        let offers = await product.getEligiblePromotionalOffers()
+        expect(offers).to(haveCount(1))
+        let offer = try XCTUnwrap(offers.first)
+
+        _ = try await Purchases.shared.purchase(product: product, promotionalOffer: offer)
+
+        let entitlement2 = try self.verifyEntitlementWentThrough()
+//        expect(entitlement.periodType) == .trial
+
+        print(entitlement2)
```

Comment:
@aboedo I've tried different types of promo offer and they all make the entitlement unaffected. Would you be able to help me debug this next week? It's possible that this is exposing something's wrong.

Reply:
Happy to help! let's chat during the week

Reply:
🤔 thinking about this, I don't think we have a specific field in the backend response that would tell you whether the promotionalOffer was applied. 
However, maybe we can figure out a good proxy for it? 

Like if the product id remains the same but you made a new purchase, then you _must_ have gone through a subscription offer. This is because you wouldn't be able to re-purchase the same product. 

We can try to figure out the "new purchase" bit by checking that the purchase date (not the original one, the one for the new purchase) is different than before. 

Would that work?

Reply:
> Like if the product id remains the same but you made a new purchase, then you must have gone through a subscription offer. This is because you wouldn't be able to re-purchase the same product.

Makes sense. In that way, this test is already checking that. But it would be nice to have something more explicit.

> We can try to figure out the "new purchase" bit by checking that the purchase date (not the original one, the one for the new purchase) is different than before.

Yeah, but I think the date wasn't changing? Maybe it's because they happened so quickly, let me check one more time though.